momentum/4/lua/ref-msys-tls-params-set.md

```diff
@@ -0,0 +1,54 @@
+---
+title: "msys.tls_params.set"
+description: "msys tls_params set Set a tls parameter string in the Momentum configuration, overriding if the parameter is already set"
+---
+
+<a name="lua.ref.msys.tls_params.set"></a> 
+## Name
+
+msys.tls_params.set — Set a tls parameter string in the Momentum configuration, overriding if the parameter is already set.
+
+## Synopsis
+
+`require('msys.tls_params');`
+
+`msys.tls_params.set(tls_params, "param", "val");`
+
+```
+tls_params: mixed
+param: string
+val: string
+```
+
+## Description
+
+This function sets a tls parameter string in the Momentum configuration, overriding if the parameter is already set.
+
+<a name="lua.ref.msys.tls_params.set.example"></a> 
+
+The following example shows how to use this function in the `get_outbound_tls_parameters` hook to override all outbound tls parameters.
+It is important to note that the use of this hook will remove all outbound tls configs, requring the user to overwrite them.
```

Comment:
I think it's worth calling out that this function is only valid to call on the `params` parameter to `get_outbound_tls_parameters` and `get_inbound_tls_parameters`.

If you tried calling this in some other context I believe what would happen is it would overwrite whatever value was on the lua stack at that position with these strings.....which I think could cause bad things to happen.

I wonder if it would make more sense to document the use of this inline with how to use `get_outbound_tls_params` and `get_inbound_tls_parameters` - but it seems we don't even document those at all?

Reply:
And this warning isn't exactly true I don't think is it?  It looks to me like the tls params object is unique per connection - so you could do something like:

```
if msg.routing_domain == "whatever"
   # set unique params just for that domain
end
```

Right?  it might be worth throwing a test in for that in the other PR

Reply:
We started with the info in the escalation linked in the ticket, and I thought Matt's testing confirmed it, which said calling `get_outbound_tls_params` resets the string values in `params`. Can we add back the test showing the reset behavior?
```suggestion
It is important to note that the use of this hook will reset outbound tls config for the current connection, requiring the user to set a value for all keys listed in the example below.
```

Reply:
👍 - yea I just threw a test together and it definitely seems to be *per connection* to me.  So I agree that we should rephrase it like above to more clearly explain that.  This specific example certainly will configure these SSL params for *all* connections, but it doesn't have to be the case.